biogeochem/EDCanopyStructureMod.F90

```diff
@@ -1625,12 +1633,10 @@ subroutine leaf_area_profile( currentSite )
                    ! is obscured by snow.
                    
                    layer_top_height = currentCohort%height - &
-                        ( real(iv-1,r8)/currentCohort%NV * currentCohort%height *  &
-                        prt_params%crown_depth_frac(currentCohort%pft) )
-                   
+                        ( real(iv-1,r8)/currentCohort%NV * crown_depth )
+                                           
                    layer_bottom_height = currentCohort%height - &
-                        ( real(iv,r8)/currentCohort%NV * currentCohort%height * &
-                        prt_params%crown_depth_frac(currentCohort%pft) )
+                        ( real(iv,r8)/currentCohort%NV * crown_depth )
```

Comment:
After discussing this with @rgknox, I ran a test manually reverting this section of code to a form that is more consistent with the old method like so:
```
 layer_top_height = currentCohort%height - &
                        ( real(iv-1,r8)/currentCohort%NV * currentCohort%height *  &
                          prt_params%allom_h2cd1(currentCohort%pft) )
```
Doing so resulted in b4b comparisons against a baseline built with Jessie's default allometry updates from #1093.

@rgknox insight here was recalling that we have had similar issues with attempting to refactor this section of code in the past.  As such, we feel that the code as written is good to go and the roundoff diffs are acceptable.

Reply:
For posterity I should note that https://github.com/NGEET/fates/issues/899 as a related connection here.